pkg/machinery/config/encoder/encoder.go

```diff
@@ -192,33 +212,54 @@ func toYamlNode(in interface{}, flags CommentsFlags) (*yaml.Node, error) {
 				fieldDoc = getDoc(value)
 			}
 
-			if !defined && flags.enabled(CommentsExamples) {
-				example := renderExample(fieldName, fieldDoc, flags)
+			// inlineExample is reandered after the value
+			var inlineExample string
 
-				if example != "" {
-					examples = append(examples, example)
-					skip = true
+			if empty && flags.enabled(CommentsExamples) && fieldDoc != nil {
+				if !skip {
+					fieldDocCopy := *fieldDoc
+					fieldDocCopy.Comments = [3]string{"", "", ""}
```

Comment:
I guess that should do the same thing:

```suggestion
					fieldDocCopy.Comments = [3]string{}
```